src/TimeRestricted.sol

```diff
@@ -36,15 +36,55 @@ import {BytesHelper} from "src/BytesHelper.sol";
 /// and the actual module addresses are the same
 /// check that the owners are the same
 
+/// Blocks all delegate calls, as the owners and modules could be changed.
+/// Does not allow changing of the implementation contract either through
+/// a normal safe transaction.
+/// The implementation contract can still be upgraded through the timelock
+/// using module calls back into the safe with a delegatecall.
```

Comment:
Using which module? Can u please explain how this will work, I think If timelock makes a delegate call back to safe, the storage slots of timelock will be updated and not the safe.

Reply:
Ah, got it. Nice! But the upgrade will only be possible through Timelock so that's good.